syntax = "proto3";


service RawFilesService {
  rpc CreateRecordRawFiles (CreateRequestRawFiles) returns (CreateResponseRawFiles);
  rpc ReadRecordRawFiles (ReadRequestRawFiles) returns (ReadResponseRawFiles);
  rpc UpdateRecordRawFiles (UpdateRequestRawFiles) returns (UpdateResponseRawFiles);
  rpc DeleteRecordRawFiles (DeleteRequestRawFiles) returns (DeleteResponseRawFiles);
}

message CreateRequestRawFiles {
  int64 project_id = 1;
  bytes data = 2;
  int32 status = 3;
}

message CreateResponseRawFiles {
  bool success = 1;
  string message = 2;
}


message ReadRequestRawFiles {
  int64 id = 1;
  int64 project_id = 2;
  bytes data = 3;
  int32 status = 4;
}

message ReadResponseRawFiles {
  int64 id = 1;
  int64 project_id = 2;
  bytes data = 3;
  string timestamp = 4;
  int32 status = 5;
  bool success = 6;
  string message = 7;
}


message UpdateRequestRawFiles {
  int64 id = 1;
  int64 project_id = 2;
  bytes data = 3;
  int32 status = 5;
  bool success = 6;
  string message = 7;
}

message UpdateResponseRawFiles {
  bool success = 1;
  string message = 2;
}


message DeleteRequestRawFiles {
  int64 id = 1;
  int64 project_id = 2;
  bytes data = 3;
  string timestamp = 4;
  int32 status = 5;
  bool success = 6;
  string message = 7;
}

message DeleteResponseRawFiles {
  bool success = 1;
  string message = 2;
}